dask/tests/test_config.py

```diff
@@ -297,3 +298,43 @@ def test_expand_environment_variables(inp, out):
         assert expand_environment_variables(inp) == out
     finally:
         del os.environ['FOO']
+
+
+@pytest.mark.parametrize('key', ['custom_key', 'custom-key'])
+def test_normalize_key(key):
+    assert normalize_key(key) == 'custom-key'
+
+
+def test_normalize_nested_keys():
+    config = {'key_1': 1,
+              'key_2': {'nested_key_1': 2},
+              'key_3': 3
+              }
+    expected = {'key-1': 1,
+                'key-2': {'nested-key-1': 2},
+                'key-3': 3
+                }
+    assert normalize_nested_keys(config) == expected
+
+
+def test_env_var_normalization(monkeypatch):
+    value = 3
+    monkeypatch.setenv('DASK_A_B', value)
+    dask.config.refresh()
+
+    assert get('a_b') == value
+    assert get('a-b') == value
+
+    # Maks sure to remove this test key from config
+    dask.config.config.pop('a-b')
```

Comment:
Yeah, that's what I was thinking too. Thanks for the feedback

Reply:
OoOo actually it looks like `get` and `refresh` have `config=` keyword arguments that let you provide your own configuration mapping instead of using the global `dask.config.config`